contracts/ZachRefund.sol

```diff
@@ -1,47 +1,54 @@
 //SPDX-License-Identifier: MIT
-pragma solidity ^0.8.19;
+pragma solidity 0.8.21;
 
-import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {MerkleProof} from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
 
 contract ZachRefund is Ownable {
-    bytes32 public immutable merkleRoot;
-    IERC20 public immutable token;
+    using SafeERC20 for IERC20;
+    using MerkleProof for bytes32[];
```

Comment:
why use it this way instead of the previous way?

Reply:
I personally like `using for <T>` more since the functions from the library can be attached to certain types and thus it means you need to be explicit about it. So the reason is explicitness. You don't like it?

Reply:
i was just curious about it, no opinion about it